src/application.coffee

```diff
@@ -218,9 +218,15 @@ fetch = (app, setDeviceUpdateState = true) ->
 			Promise.join utils.getConfig('apiKey'), utils.getConfig('uuid'), (apiKey, uuid) ->
 				if conf['RESIN_SUPERVISOR_DELTA'] == '1'
 					logSystemEvent(logTypes.downloadAppDelta, app)
-					requestTimeout = checkInt(conf['RESIN_SUPERVISOR_DELTA_REQUEST_TIMEOUT'], positive: true) ? 30 * 60 * 1000
-					totalTimeout = checkInt(conf['RESIN_SUPERVISOR_DELTA_TOTAL_TIMEOUT'], positive: true) ? 24 * 60 * 60 * 1000
-					dockerUtils.rsyncImageWithProgress(app.imageId, { requestTimeout, totalTimeout, uuid, apiKey }, onProgress)
+					deltaOpts = {
+						uuid, apiKey
+						# use user-defined timeouts, but fallback to defaults if none is provided.
+						requestTimeout: checkInt(conf['RESIN_SUPERVISOR_DELTA_REQUEST_TIMEOUT'], positive: true)
+						applyTimeout: checkInt(conf['RESIN_SUPERVISOR_DELTA_APPLY_TIMEOUT'], positive: true)
```

Comment:
@pcarranzav, as it is now, if any of the options aren't supplied, the variables will be undefined and will cause `resumable-request` to fall back to defaults, which are set here: https://github.com/resin-io/resin-supervisor/pull/483/files#diff-5b6270202e7b362597737840968ab7fcR23